### PR TITLE
Add negative tests for RestAPiCreation test and APIrequest constructor

### PR DESCRIPTION
**Pull request by kavithasub**
## Purpose
- Add negative test methods for RestApiCreation test case
- Add override ApiRequest constructor

### CODE REVIEW

File: product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java [hunk 1]
```diff
@@ -35,6 +41,23 @@ public APIRequest(String name, String context, String visibility, String version
...
     }
 
+    public APIRequest(String name, String context, String visibility, String version, String resource, String tiersCollection, URL endpointUrl) {
+        this.name = name;
+        this.context = context;
+        this.visibility = visibility;
+        this.version = version;
+        this.resource = resource;
+        this.tiersCollection = tiersCollection;
+        try{
```
Comment (Krishanx92) [inline product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Formatting issue.

File: product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java [hunk 2]
```diff
@@ -35,6 +41,23 @@ public APIRequest(String name, String context, String visibility, String version
...
+            this.endpoint =
+                    new JSONObject("{\"production_endpoints\":{\"url\":\""
+                            + endpointUrl + "\",\"config\":null},\"endpoint_type\":\""
+                            + "http" + "\"}");
+        }catch (JSONException e){}
```
Comment (Krishanx92) [inline product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Formatting issue.

File: product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java [hunk 3]
```diff
@@ -49,6 +72,14 @@ public void init() {
         this.addParameter("visibility", this.visibility);
         this.addParameter("version", this.version);
         this.addParameter("swagger", this.swagger);
+        if(tiersCollection != null)
+        {
```
Comment (Krishanx92) [inline product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Shall we fix formatting issues

File: product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java (hunk 1 above)
Comment (kavithasub) [reply product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Fix it in dc9a5c9555745294c47d4dc0e6cad63bdf5e4410

File: product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java (hunk 2 above)
Comment (kavithasub) [reply product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Fixed in dc9a5c9555745294c47d4dc0e6cad63bdf5e4410

File: product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java (hunk 3 above)
Comment (kavithasub) [reply product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api-and-apply-qos/common-utils/src/main/java/org/wso2/am/scenario/test/common/APIRequest.java]: Fix it in dc9a5c9555745294c47d4dc0e6cad63bdf5e4410

**Pull request merged**